mws/utils/params.py

```diff
@@ -0,0 +1,231 @@
+"""Parameter manipulation utilities."""
+
+from collections.abc import Iterable, Mapping
+from urllib.parse import quote
+import datetime
+import json
+
+from mws.errors import MWSError
+
+
+def enumerate_param(param, values):
+    """Builds a dictionary of an enumerated parameter, using the param string and some values.
+    If values is not a list, tuple, or set, it will be coerced to a list
+    with a single item.
+
+    Example:
+        enumerate_param('MarketplaceIdList.Id', (123, 345, 4343))
+    Returns:
+        {
+            MarketplaceIdList.Id.1: 123,
+            MarketplaceIdList.Id.2: 345,
+            MarketplaceIdList.Id.3: 4343
+        }
+    """
+    if not isinstance(values, (list, tuple, set)):
+        # Coerces a single value to a list before continuing.
+        values = [values]
+    if not any(values):
+        # if not values -> returns ValueError
+        return {}
+    if not param.endswith("."):
+        # Ensure this enumerated param ends in '.'
+        param += "."
+    # Return final output: dict comprehension of the enumerated param and values.
+    return {"{}{}".format(param, idx + 1): val for idx, val in enumerate(values)}
```

Comment:
could we not just do `enumerate(values, 1)` to start from 1 rather than have to do `idx + 1` ?

Reply:
Learned something new today! I'll get on that. 👍